ClusterHealthIT.testHealthOnMasterFailover failures

this failed for me once, but I can see 4 other failures during the month.
It does not reproduce to me either. I suspect this might be same race condition in a test?

build stats
https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-60d,mode:quick,to:now))&_a=(columns:!(_source),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testHealthOnMasterFailover),sort:!(process.time-start,desc))

gradle failure
https://gradle-enterprise.elastic.co/s/pbjvtw7h4rvue/tests/kyv2y2z3r4v7m-fatnnfmmekcxc

reproduce line
`./gradlew ':server:integTest' --tests "org.elasticsearch.cluster.ClusterHealthIT.testHealthOnMasterFailover" -Dtests.seed=79E33053A1817125 -Dtests.security.manager=true -Dtests.locale=en-IN -Dtests.timezone=Asia/Dhaka -Dcompiler.java=13
`

stacktrace
```
:server:integTest » org.elasticsearch.cluster.ClusterHealthIT » testHealthOnMasterFailover (13.423s)
expected same:<RED> was not:<GREEN>
java.lang.AssertionError: expected same:<RED> was not:<GREEN>Close stacktrace
at __randomizedtesting.SeedInfo.seed([79E33053A1817125:E70C1F7EA26B6DC3]:0)
at org.junit.Assert.fail(Assert.java:88)
at org.junit.Assert.failNotSame(Assert.java:828)
at org.junit.Assert.assertSame(Assert.java:771)
at org.junit.Assert.assertSame(Assert.java:782)
at org.elasticsearch.cluster.ClusterHealthIT.testHealthOnMasterFailover(ClusterHealthIT.java:310)
```

@original-brownbear would you be able to take a quick look?

--- a/server/src/test/java/org/elasticsearch/cluster/ClusterHealthIT.java
+++ b/server/src/test/java/org/elasticsearch/cluster/ClusterHealthIT.java
@@ -303,7 +303,8 @@ public class ClusterHealthIT extends ESIntegTestCase {
         // Run a few health requests concurrent to master fail-overs against a data-node to make sure master failover is handled
         // without exceptions
         for (int i = 0; i < 20; ++i) {
-            responseFutures.add(client(node).admin().cluster().prepareHealth().setWaitForEvents(Priority.LANGUID).execute());
+            responseFutures.add(client(node).admin().cluster().prepareHealth().setWaitForEvents(Priority.LANGUID)
+                .setWaitForGreenStatus().execute());
             internalCluster().restartNode(internalCluster().getMasterName(), InternalTestCluster.EMPTY_CALLBACK);
         }
         for (ActionFuture<ClusterHealthResponse> responseFuture : responseFutures) {

